### ISSUE TITLE
Apples new open source project. 

### ISSUE THREAD

**Issue by milesmcclane** [strongbox-password-safe/Strongbox#349]
Any thoughts on this? Could be useful to implement. 

https://github.com/apple/password-manager-resources

**Comment by strongbox-mark** [comment]
Yes @milesmcclane, thanks! 

This looks like it could be quite useful in a few places... A few features I can think of:

- **Generation** - If URL is known and the site has a quirk then use the ruleset instead of your configured generation mode
- **Change Password** - If the site is known (from the URL) offer a change password link in the Password field
- **URL/Host Alternatives** - If the site (determined from the URL, let's say https://amazon.com), let's say is known to use shared credentials (e.g. amazon.com and amazon.co.uk), then when the user visits say amazon.co.uk in Safari, present the Amazon.com entry as a QuickType option? Possibly also in regular Autotype search...

I think one of the things I'd really like to see from Apple is a better Auto-Fill interface for creating passwords. It should be extended to allow Auto-Fill providers (e.g. Strongbox) to create and store a password directly inline in App much like QuickType. I might just raise this on that repo and see if they can help/point me in the right direction.

We can keep this here for further discussion and then breakout when there is time for development on this. Probably worth letting that repo settle too as there are a lot of changes going in at the moment.

**Comment by georgesnow** [comment]
I agree with the quicktype providers to providers the inline option for create and store, but also a password recipes when creating the entry. I thought about opening ticket for same thing. 

Maybe this is a sign 10.16 will provide Autofill/Quicktype like feature for macOS for 3rd party password managers...

**Comment by rmondello** [comment]
> Maybe this is a sign 10.16 will provide Autofill/Quicktype like feature for macOS for 3rd party password managers...

I wouldn’t say it was a sign, but that did happen!

Things have settled down in the repo compared to when it launched, but we’re consistently incorporating changes. Feel free to reach out any time!

**Comment by georgesnow** [comment]
https://developer.apple.com/documentation/authenticationservices/ascredentialidentitystore

Ah very cool I see it’s added for macos 11 beta.

**Comment by strongbox-mark** [comment]
That's great, really appreciate the contact @rmondello 

@georgesnow Thanks for the link - Looks like it should be a straightforward port over to Mac from the iOS code.

**Comment by georgesnow** [comment]
Curious if once Big Sur (11.0) will this become more of a possibility for the Mac app? Also this would I believe would resolve issue #21 at least for safari users.

**Comment by strongbox-mark** [comment]
Yes, #21 should be fixed automatically once the Password AutoFill extension is available on Big Sur... Might take another few weeks as I'm having some trouble getting it running on Big Sur right now.
